src/error.rs

```diff
@@ -94,6 +94,12 @@ pub type Result<T> = std::result::Result<T, Error>;
 // `wolfSSL_get_error`, which returns a `c_int`
 fn wolf_error_string(raw_err: std::ffi::c_ulong) -> String {
     let mut buffer = vec![0u8; wolfssl_sys::WOLFSSL_MAX_ERROR_SZ as usize];
+    // SAFETY:
+    // The underlying `C` api copies the static c string using strncpy with maximum s↴ize `WOLFSSL_MAX_ERROR_SZ`
+    // So we create a `mutable` buffer with size `WOLFSSL_MAX_ERROR_SZ` to satify the api req
+    // https://github.com/wolfSSL/wolfssl/blob/v5.6.3-stable/src/ssl.c#L4560
+    // https://github.com/wolfSSL/wolfssl/blob/v5.6.3-stable/src/internal.c#L23628
```

Comment:
I think the first/primary focus of the safety comments should be the wolfssl documentation.

The comments should only need to delve into references to specific code if the documentation is lacking/imprecise/ambiguous.

The wolfSSL people are IMHO at liberty to change the implementation in arbitrary ways, but the API docs are their contract with us.

So here I'd expect primarily to reference: https://www.wolfssl.com/doxygen/group__Debug.html#ga91d8474ba8abcf3fe594928056834993 and/or https://www.wolfssl.com/documentation/manuals/wolfssl/group__Debug.html#function-wolfssl_err_error_string not the code.
e.g.

```rust
/// SAFETY:
/// [`wolfSSL_ERR_error_string()`][0] is documented to store at most `WOLFSSL_MAX_ERROR_SZ` bytes, so `buffer` is appropriately sized.
///
/// [0]: https://www.wolfssl.com/doxygen/group__Debug.html#ga91d8474ba8abcf3fe594928056834993
```

(An aside on the implementation: On failure a static string is returned describing the problem -- we don't deal with that case here, probably we should).